wordpress-mobile/WordPress-Android
# Stop using global scope in coroutines

Fixes #9527

Most of the changes in this PR should be obvious. The global scope is by default on the main thread so I've replaced it with the main dispatcher. The behaviour of the changes should be exactly as it was before. 

This also includes the FluxC changes from https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1536

To test:
- Insights:
  - Try to add/remove blocks in the Insights management screen
  - Blocks are added/removed when you save the changes
  - Move a block up and down in the Insights
  - The Publish activity block is shown correctly
- Stats: 
  - The bar chart is visible and updates the screen when you tap on a bar
  - The country block is visible
  - Widgets load data as expected for sites that haven't loaded data in the app yet
- Actions in Pages work (set a parent, move to draft/delete)
- Making the comment go full screen opens the keyboard

PR submission checklist:
